### ISSUE TITLE
create a new artifact for future projects

### ISSUE THREAD

**Issue by SylvainJanet** [SylvainJanet/personal-website-app#18]
once every other previous issue is resolved, the app is ready to be a nice basic app for future project. Indeed :
- CI/CD pipelines are complete, effective, efficient and quite exhaustive to form a basis for development
    - app is tested
    - code is linted
    - sonarcloud analysis is performed
    - app is built
    - app is deployed once all the other jobs pass
- 3 environments are setup
    - one for dev, one for staging and one for prod. 
        - Dev = local development. Uses local DB. 
        - Staging = on VPS but deployed on a restricted domain not available for the public. Uses a distant DB but different than the one containing crucial data. Acts mostly as a proxy, to ensure everything works well when the app is deployed on the VPS. 
        - Prod = production, on VPS, accessible for the public. DB has the real data that has to be protected and should not be messed with during development and deployment.
- one entity is used with a few tables in the DB
    - used to store text content in different languages.
    - repo, service, controller are setup
    - CRUD and basic methods are setup
- every parameter of the app is localized in a final class containing static final fields so that it's easy to tweak some parameters
- the app is tested
    - the tests are integrated in quality gates in CI/CD pipelines
    - TDD approach should be used, since this artifact already has minimal functionality
- the app is documented
  - and the documentation is part of the CI/CD process. Every new code should be documented directly, but the documentation generation is already part of the pipeline
- the DB changes are ready to be tracked
    - after the first DB creation, structural changes are taken care of and tracked. Moreover, data corruption can be prevented
- admin login is ready
    - the app is ready to support having one (or some) basic admin
    - the app does not have user creation, but permissions are set up and it could be done. 
        - I do not think every app has to have users so it's not part of this artifact. And even if an app does, there are many ways of conceiving how user should connect to the app. Is it based on an email account ? Linked to a social Facebook/Google ? What field does the entity user have ? I think those questions will be app specific for my future projects and hence a generic answer should not be given in this app
- error management : the app properly displays errors of different type, so that the user might distinguish between some errors that have been anticipated and are part of the expected behaviour of the app (for instance, trying to get a string with an incorrect selector)  or if it is part of the unexpected behaviour (which, in the best case scenario, should be because of bugs so that every error should be clearly communicated to the user, not just an error 500 for instance)
- security review has been done and the most commonplace security issues have been dealt with, to some extent.

With all of that said, this app should be a great artifact to start my next projects, having most of the basic stuff taken care of. Only app-specific issues and code have to be produced from this point on.
/!\ Security review and most importantly dependency checks should be done, especially if this artifact is used in the long future, some stuff that are currently (as of time of release) safe might not be in the future.
